---
title: "RNA-seq FAQs"
date: 2025-06-26
date-modified: 2025-06-26
---

## `nfcore/rnaseq` vs `nfcore/differentialabundance`

`nfcore/rnaseq` is used for pre-processing, specifically generating a count table from the `FASTQ` files. `nfcore/differentialabundance` is then used for the differential expression/abundance analysis, that is comparing expression of genes across different conditions.

## Count file contains floating point numbers

The count file is generated by `nfcore/rnaseq` using the default path (aligner: `STAR`, quantification by: `Salmon`). `Salmon` "infers" (aggregates) gene counts (actually count estiamtes) from transcript-level count (estimates) and takes gene lengths into account (the gene lengths might differ across samples since not all isoforms could be expressed in a given samples). All in all the resulting count matrix contains floating point numbers as these are just estimates. One can simply `round()` if feeding the count table to `DESeq2`, for example `nfcore/differentialabundance` [rounds the count matrix](https://github.com/nf-core/differentialabundance/blob/3dd360fed0dca1780db1bdf5dce85e5258fa2253/modules/nf-core/deseq2/differential/templates/deseq_de.R#L339).

## DESeq2 normalization

`DESeq2` uses [median of ratios](https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html) for normalization.